Framework/Core/src/DataSpecUtils.cxx

```diff
@@ -54,10 +54,8 @@ void DataSpecUtils::describe(char* buffer, size_t size, InputSpec const& spec)
     origin[4] = 0;
     char description[17];
     description[16] = 0;
-    snprintf(buffer, size, "%s/%s/%" PRIu64,
-             (strncpy(origin, concrete->origin.str, 4), origin),
-             (strncpy(description, concrete->description.str, 16), description),
-             concrete->subSpec);
+    snprintf(buffer, size, "%s/%s/%" PRIu32, (strncpy(origin, concrete->origin.str, 4), origin),
```

Comment:
Out of curiosity, doesn't C++ have a trait to select `PRI*`s?

Reply:
aside from some magical invocations of cout to a streambuf? no idea... maybe boost has something more robust.

Reply:
I meant more something which does:

```C++
template <typename T>
struct pri_traits {
};

template <>
struct pri_traits<uint32_t> {
   constexpr decltype(PRI32) value = PRIu32; 
};

snprintf(buffer, size, "%s/%s/%" pri_traits<SubSpecificationType>::value);
```

now that I think about it, probably the fact that PRIu32 is some string literal is what does not allow it... Oh, well...